x/compute/internal/keeper/secret_contracts_test.go

```diff
@@ -185,7 +209,15 @@ func execHelper(t *testing.T, keeper Keeper, ctx sdk.Context, contractAddress sd
 }
 
 func initHelper(t *testing.T, keeper Keeper, ctx sdk.Context, codeID uint64, creator sdk.AccAddress, initMsg string, isErrorEncrypted bool, gas uint64) (sdk.AccAddress, []ContractEvent, cosmwasm.StdError) {
-	initMsgBz, err := wasmCtx.Encrypt([]byte(initMsg))
+
+	hash := keeper.GetCodeInfo(ctx, codeID).CodeHash
+
+	msg := types.SecretMsg{
+		CodeHash: []byte(hex.EncodeToString(hash)),
+		Msg:      []byte(initMsg),
+	}
+
+	initMsgBz, err := wasmCtx.Encrypt(msg.Serialize())
```

Comment:
This pattern repeats a lot. can it maybe be a helper function? or a part of `wasmCtx.Encrypt`?